src/integration-test/utils.ts

```diff
@@ -188,3 +152,47 @@ export function withPing<T extends object>(stream: Observable<T>, interval = 100
     .filter(event => typeof event === 'object')
     .map(event => event as T);
 }
+
+export function saveToCache(cacheDir: string, cacheFileName: string) {
+  return (clients: Partial<CharlesClients>) => {
+    try {
+      mkpath.sync(cacheDir);
+    } catch (error) {
+      // nothing
+    }
+    const cacheFile = path.join(cacheDir, cacheFileName);
+    const clientDtos = Object.keys(clients)
+      .reduce((p: any, tt: TeamType) => ({ ...p, [tt]: clients[tt]!.toDto() }), {});
+    fs.writeFileSync(cacheFile, JSON.stringify(clientDtos, undefined, 2));
+  };
+}
+
+export function loadFromCache(cacheDir: string, cacheFileName: string): Partial<CharlesClients> {
+  const cacheFile = path.join(cacheDir, cacheFileName);
+  const clientDtos = JSON.parse(fs.readFileSync(cacheFile).toString());
+  return Object.keys(clientDtos)
+    .reduce((p: any, tt: TeamType) => ({ ...p, [tt]: CharlesClient.load(clientDtos[tt]) }), {});
+}
+
+export function getAnonymousClient(client: CharlesClient) {
+  const anonymous = new CharlesClient(client.url, '');
+  anonymous.teamId = 9999999;
```

Comment:
It's possible to create the client without these properties. But we anyway need to come up with a solution on how to request entities which we know won't exist. This has room for improvent.